codec/src/test/java/io/netty/handler/codec/ByteToMessageDecoderTest.java

```diff
@@ -316,6 +313,7 @@ public ByteBuf writeBytes(ByteBuf src) {
                 throw error;
             }
         };
+        cumulation.writeByte(1);
```

Comment:
The test assumes that MERGE_CUMULATOR will write some bytes and throw an exception, but the new behaviour (which is completely permitted by the API) is to return the right hand buffer if the left is empty.  To exercise the intended behaviour of the test, we just make sure the buffer is non-empty.

We could if you prefer change the behaviour of the default cumulators to only return the right hand side if the left hand buffer is both empty and non-writable, but this would still seem to make little sense for the COMPOSITE_CUMULATOR, which seems to never ideally want to aggregate an empty buffer.  The MERGE_CUMULATOR also probably benefits from defaulting to returning the only non-empty buffer to avoid a copy, although the case is muddier if the caller is hoping a writable but empty left hand side would be written to.

Reply:
ok make sense... I just wanted to understand why this change was done.

Reply:
can you maybe add a comment ?